realm/realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -823,6 +823,21 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeSetByteArray(
     } CATCH_STD()
 }
 
+JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeSetNull(
+    JNIEnv* env, jobject, jlong nativeTablePtr, jlong columnIndex, jlong rowIndex)
+{
+    Table* pTable = TBL(nativeTablePtr);
+    if (!TBL_AND_COL_INDEX_VALID(env, pTable, columnIndex))
+        return;
```

Comment:
AFAIK, I _believe_ all `*_VALID` function family throws via `env->ThrowNew` in JNI. ( I cannot assure you the term **all** since I have not checked everything, but for those ones I've checked ended up in calling `ThrowException` in `util.cpp`. )